### PR TITLE
Feature/b database modeling

### PR DESCRIPTION
**Pull request by Pellumbengineer**
This is the final DB model drawing.

This is the link to the uml diagram -> https://drive.google.com/file/d/108J6exfhiWS5OFqtvQM77nalTc_Nw13K/view?usp=sharing

dbModeling

``` 
//institutions
{
   id:1
   name:TUM
   country:Germany
   profilePictureURL: s3_bucket_link
   backgroundPictureURL: s3_bucket_link
   adminIds: [1],
   invitedMemberIds: [],
   visibility: PUBLIC
}

//projects
{
   id: 1
   title:Biotechnology Project
   description: Biotechnology Project bla bla
   adminIds: [1],
   invitedMemberIds: [],
   memberIds: [2],
   visibility: PUBLIC,
   institutionId: 1
}

{
   id: 2
   title: JST Project
   description: JST Project bla bla
   adminIds: [1],
   invitedMemberIds: [],
   memberIds: [2],
   visibility: PRIVATE,
   institutionId: empty
}

// users
{
    id: 1
    firstName: Pellumb;
    lastName: Baboci;
    email: pellumb@example.com;
    password?: 123456;
    note: string;
    token: string;
    emailVerificationToken: string;
    isEmailVerified: true
    isAdministrator: true;
}

{
    id: 2
    firstName: TesUser;
    lastName: user;
    email: test@example.com;
    password?: 123456;
    note: string;
    token: string;
    emailVerificationToken: string;
    isEmailVerified: true
    isAdministrator: false;
}

//projectJobs
{
   id: 1
   projectId: 1
   ...
   ...
}

.....

//tokes
{
   ...
   ...
}

.....

//password_reset_token
{
   ...
   ...
}

.....

```

### CODE REVIEW

File: Backend/src/database/models/institution.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import {Document, model, Schema} from "mongoose";
+
+export interface IInstitution extends Document {
+   name:string,
+   country:string,
+   profilePictureURL: string,
+   backgroundPictureURL: string,
+   adminIds: [Schema.Types.ObjectId],
+   invitedMemberIds: [Schema.Types.ObjectId],
+   projects: [Schema.Types.ObjectId],
```
Comment (Eronavrp) [inline Backend/src/database/models/institution.ts:10]: Shouldn't we here also include memberIds not just ones who are invited.

File: Backend/src/API/REST/routes/project/projectRouter.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            const admin_user_id = req.user_id;
+
+            if (!(title && description && visibility && admin_user_id))
+                return res.status(400).send("Missing parameters");
+            
+            if(!(Object.values(visibilityStatus).includes(visibility)) )
+                return res.status(400).send("Visibility parameter is wrong format. You should type one of the followings: PRIVATE, PUBLIC, BY_INSTITUTION");
+
+            if (await projectModel.findOne({title}))
+                return res.status(409).send("Project with the given name already exists!");
```
Comment (Eronavrp) [inline Backend/src/API/REST/routes/project/projectRouter.ts:23]: What if projects belong to different institutions, should this also hold true

File: Backend/src/database/models/institution.ts (hunk 1 above)
Comment (Pellumbengineer) [reply Backend/src/database/models/institution.ts:10]: yes, member ids field will be added.

File: Backend/src/API/REST/routes/project/projectRouter.ts (hunk 2 above)
Comment (Pellumbengineer) [reply Backend/src/API/REST/routes/project/projectRouter.ts:23]: Good point, that is exactly the case. This need to be changed.

**Pull request merged**